hamcrest/Hamcrest/Arrays/SeriesMatchingOnce.php

```diff
@@ -31,7 +31,7 @@ public function matches($item)
     public function isFinished()
     {
         if (!empty($this->_elementMatchers)) {
-            $nextMatcher = current($this->_elementMatchers);
+            $nextMatcher = reset($this->_elementMatchers);
```

Comment:
The `current` will get currently iterated element in the array (the iteration might happened before this method was even called). The `reset` will reset current element index in the array and therefore will always get you 1st element in the array.

Fact, that tests for the `isFinished` method even passed proves, that they are incomplete (e.g. don't test case when we have 2+ elements in `$this->_elementMatchers`).

Also I wonder under which conditions the `current` will return boolean. Maybe the `$this->_elementMatchers` array was corrupted.